majifix311/src/main/java/com/example/majifix311/ui/ProblemListActivity.java

```diff
@@ -1,22 +1,147 @@
 package com.example.majifix311.ui;
 
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
 import android.os.Bundle;
-import android.os.PersistableBundle;
 import android.support.annotation.Nullable;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v4.app.Fragment;
+import android.support.v4.content.LocalBroadcastManager;
 import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Toast;
 
+import com.example.majifix311.EventHandler;
 import com.example.majifix311.R;
+import com.example.majifix311.api.ReportService;
+import com.example.majifix311.models.Problem;
+import com.example.majifix311.utils.Utils;
+
+import java.util.ArrayList;
 
 /**
  * This activity shows a list of reported issues.
  */
 
-public class ProblemListActivity extends AppCompatActivity {
+public class ProblemListActivity extends AppCompatActivity implements ErrorFragment.OnReloadClickListener {
+    @Utils.UiState
+    private String mUiState = Utils.NONE;
+    private int mFragmentContainerRes = R.id.frl_fragmentContainer;
+    private FloatingActionButton mFab;
+
+    private BroadcastReceiver mMyReportedProblemsReceived = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            if (intent.getBooleanExtra(EventHandler.IS_SUCCESS, false)) {
+                ArrayList<Problem> problems = intent.getParcelableArrayListExtra(EventHandler.REQUEST_LIST);
+                System.out.println("Problems Received! " + (problems == null ? "null" : problems.size()));
+                if (problems == null || problems.isEmpty()) {
```

Comment:
I'm not sure how you are implementing it. For example: if there is still a preliminary flag, then this can respond to the preliminary flag. I also have a thought that maybe a empty response from the db should be handled by the service and not be sent to the ui, as there is no ui update in that case (If db empty comes first the loading bar should continue spinning, and if db empty comes last no list should remain).